internal/attestation/measurements/measurements.go

```diff
@@ -94,58 +70,207 @@ func (m *M) FetchAndVerify(ctx context.Context, client *http.Client, measurement
 
 // CopyFrom copies over all values from other. Overwriting existing values,
 // but keeping not specified values untouched.
-func (m M) CopyFrom(other M) {
+func (m *M) CopyFrom(other M) {
 	for idx := range other {
-		m[idx] = other[idx]
+		(*m)[idx] = other[idx]
 	}
 }
 
 // EqualTo tests whether the provided other Measurements are equal to these
 // measurements.
-func (m M) EqualTo(other M) bool {
-	if len(m) != len(other) {
+func (m *M) EqualTo(other M) bool {
+	if len(*m) != len(other) {
 		return false
 	}
-	for k, v := range m {
-		if !bytes.Equal(v, other[k]) {
+	for k, v := range *m {
+		otherExpected := other[k].Expected
+		if !bytes.Equal(v.Expected[:], otherExpected[:]) {
+			return false
+		}
+		if v.WarnOnly != other[k].WarnOnly {
 			return false
 		}
 	}
 	return true
 }
 
-// MarshalYAML overwrites the default behaviour of writing out []byte not as
-// single bytes, but as a single base64 encoded string.
-func (m M) MarshalYAML() (any, error) {
-	base64Map := make(map[uint32]string)
+// GetEnforced returns a list of all enforced Measurements,
+// i.e. all Measurements that are not marked as WarnOnly.
+func (m *M) GetEnforced() []uint32 {
+	enforced := []uint32{}
+	for idx, measurement := range *m {
+		if !measurement.WarnOnly {
+			enforced = append(enforced, idx)
+		}
+	}
+	return enforced
+}
 
-	for key, value := range m {
-		base64Map[key] = base64.StdEncoding.EncodeToString(value[:])
+// SetEnforced sets the WarnOnly flag to true for all Measurements
+// that are NOT included in the provided list of enforced measurements.
+func (m *M) SetEnforced(enforced []uint32) {
```

Comment:
what if we have an index in the enforced list that's not in m?

Reply:
Currently does nothing.
I would opt to return an error in that case